.github/workflows/_release.yml

```diff
@@ -23,60 +29,71 @@ jobs:
-      - run: zig build
-      - run: zip -j fastfec-linux-x86_64-${{ inputs.version }}.zip zig-out/bin/fastfec
-      - run: mv zig-out/lib/libfastfec.so libfastfec-linux-x86_64-${{ inputs.version }}.so
-      - run: zig build -Dtarget=aarch64-linux
-      - run: zip -j fastfec-linux-aarch64-${{ inputs.version }}.zip zig-out/bin/fastfec
-      - run: mv zig-out/lib/libfastfec.so libfastfec-linux-aarch64-${{ inputs.version }}.so
-      - run: zig build -Dwasm
-      - run: mv zig-out/lib/fastfec.wasm libfastfec-${{ inputs.version }}.wasm
-      # Mac
-      - run: zig build -Dtarget=x86_64-macos
-      - run: zip -j fastfec-macos-x86_64-${{ inputs.version }}.zip zig-out/bin/fastfec
-      - run: mv zig-out/lib/libfastfec.dylib libfastfec-macos-x86_64-${{ inputs.version }}.dylib
-      - run: zig build -Dtarget=aarch64-macos
-      - run: zip -j fastfec-macos-aarch64-${{ inputs.version }}.zip zig-out/bin/fastfec
-      - run: mv zig-out/lib/libfastfec.dylib libfastfec-macos-aarch64-${{ inputs.version }}.dylib
-      # Windows
-      - run: zig build -Dtarget=x86_64-windows
-      - run: zip -j fastfec-windows-x86_64-${{ inputs.version }}.zip zig-out/bin/fastfec.exe
-      - run: mv zig-out/lib/fastfec.dll libfastfec-windows-x86_64-${{ inputs.version }}.dll
-      - run: zig build -Dtarget=aarch64-windows
-      - run: zip -j fastfec-windows-aarch64-${{ inputs.version }}.zip zig-out/bin/fastfec.exe
-      - run: mv zig-out/lib/fastfec.dll libfastfec-windows-aarch64-${{ inputs.version }}.dll
-      - uses: actions/upload-artifact@v2
+      - name: Build target
+        if: matrix.os != 'wasm'
+        run: zig build -Dtarget=${{ matrix.architecture }}-${{ matrix.os }}
+      - name: Build target
+        if: matrix.os == 'wasm'
+        run: zig build -Dwasm
+      - name: Zip output executable
+        if: matrix.os != 'wasm'
+        run: zip -j fastfec-${{ matrix.os }}-${{ matrix.architecture }}-${{ inputs.version }}.zip zig-out/bin/fastfec
+      - name: Get library output paths
+        uses: kanga333/variable-mapper@master
+        id: library
+        with:
+          key: "${{ matrix.os }}"
+          map: |
+            {
+              "linux": {
+                "ext": "so",
+                "name": "libfastfec"
+              },
+              "mac": {
+                "ext": "dylib",
+                "name": "libfastfec"
+              },
+              "windows": {
+                "ext": "dll",
+                "name": "fastfec"
+              },
+              "wasm": {
+                "ext": "wasm",
+                "name": "fastfec"
+              }
+            }
+          export_to: output
+      - name: Move output library
+        if: matrix.os != 'wasm'
+        run: mv zig-out/lib/${{ steps.library.outputs.name }}.${{ steps.library.outputs.ext }} libfastfec-${{ matrix.os }}-${{ matrix.architecture }}-${{ inputs.version }}.${{ steps.library.outputs.ext }}
+      - name: Move output library
+        if: matrix.os == 'wasm'
```

Comment:
Wasm sure seems like the dividing line between a lot of these and the source of some repition. I wonder if there's a way to consolidate either by forming the file names earlier or branching around wasm.

Reply:
Yea I think for now it's fine and still pretty minimal. There's for sure elegant ways to improve it that could be done, but it's so much better